### ISSUE TITLE
[Masthead A11Y]: None of the descendent elements with "menuitem" role is tabbable

### ISSUE THREAD

**Issue by mo-alaa** [carbon-design-system/carbon-for-ibm-dotcom#9691]
### Description

Hi, while using the L1 masthead (contained in DotcomShell @ ibmdotcom-react), we can find that the `<li role="menuitem">`  elements in masthead do not have a `tabindex="0"` attribute.

I understand that the anchor inside the list-item has a `tabindex="0"` but,
this was raised as a violation while scanning [the page](https://www.ibm.com/impact) (L1 is used in that page) with [IBM Equal Access Accessibility Checker](https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp/related).

Thanks!
![image](https://user-images.githubusercontent.com/41386480/203791522-030738ef-eedb-477f-9a75-8e80b82c01b0.png)

### Component(s) impacted

DotcomShell of ibmdotcom-react

### Carbon for IBM.com version

@carbon/ibmdotcom-react: "^1.31.1"

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

ibm.com/impact

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-l-1

### Steps to reproduce the issue (if applicable)

This can be inspected in the [docs for the masthead L1 element](https://www.ibm.com/standards/carbon/react/?path=/story/components-masthead--with-l-1) as shown in the image.
As you can see, no `tabindex="0"`
![image](https://user-images.githubusercontent.com/41386480/203792988-3ed61476-2716-49a5-bb35-469de3a0ab88.png)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by mo-alaa** [comment]
Just a heads-up. Thanks!
